apps/extension/hooks/use-posthog-analytics.ts

```diff
@@ -0,0 +1,45 @@
+import { posthog } from "posthog-js"
+import { useCallback } from "react"
+
+import type { UseFarcasterIdentityOptions } from "./use-farcaster-identity"
+
+if (process.env.PLASMO_POSTHOG_API_KEY) {
+  posthog.init(process.env.PLASMO_POSTHOG_API_KEY, {
+    api_host: process.env.PLASMO_POSTHOG_API_HOST
+  })
+}
+
+type PosthogAnalyticsReturn = Required<
+  Pick<UseFarcasterIdentityOptions, "onSignIn" | "onSignInStart" | "onSignOut">
+>
+
+export function usePosthogAnalytics(): PosthogAnalyticsReturn {
```

Comment:
Not sure if to keep this hook outside of `useFarcasterIdentity()` or move it inside so we can clean up the files.

Also it can trigger multiple events with same type and data since the hook runs in different contexts. I don't think it is a big issue.